lib/src/obsoleted/rules/member_ordering_extended_rule.dart

```diff
@@ -0,0 +1,709 @@
+        ? List<String>.from(config['order'] as Iterable)
+        : <String>[
+            'public_fields',
+            'private_fields',
+            'public_getters',
+            'private_getters',
+            'public_setters',
+            'private_setters',
+            'constructors',
+            'public_methods',
+            'private_methods',
+          ];
+
+    return order.map(parseGroup).whereNotNull().toList();
+  }
+
+  // ignore: long-method
+  static _MemberGroup? parseGroup(String group) {
+    final type = _MemberType.parse(group.split('_').lastOrNull);
+    final regExp = RegExp(
+      '(overridden_|protected_)?(private_|public_)?(static_)?(late_)?'
+      '(var_|final_|const_)?(nullable_)?(named_)?(factory_)?',
+    );
+    final result = regExp.allMatches(group.toLowerCase());
+
+    final hasGroups = result.isNotEmpty && result.first.groupCount > 0;
+    if (hasGroups && type != null) {
+      final match = result.first;
+
+      final annotation = _Annotation.parse(match.group(1)?.replaceAll('_', ''));
+      final modifier = _Modifier.parse(match.group(2)?.replaceAll('_', ''));
+      final isStatic = match.group(3) != null;
+      final isLate = match.group(4) != null;
+      final keyword = _FieldKeyword.parse(match.group(5)?.replaceAll('_', ''));
+      final isNullable = match.group(6) != null;
+      final isNamed = match.group(7) != null;
+      final isFactory = match.group(8) != null;
+
+      switch (type) {
+        case _MemberType.field:
+          return _FieldMemberGroup._(
+            isLate: isLate,
+            isNullable: isNullable,
+            isStatic: isStatic,
+            keyword: keyword,
+            annotation: annotation,
+            memberType: type,
+            modifier: modifier,
+            rawRepresentation: group,
+          );
+
+        case _MemberType.method:
+          return _MethodMemberGroup._(
+            isNullable: isNullable,
+            isStatic: isStatic,
+            annotation: annotation,
+            memberType: type,
+            modifier: modifier,
+            rawRepresentation: group,
+          );
+
+        case _MemberType.getter:
+        case _MemberType.setter:
+        case _MemberType.getterAndSetter:
+          return _GetSetMemberGroup._(
+            isNullable: isNullable,
+            isStatic: isStatic,
+            annotation: annotation,
+            memberType: type,
+            modifier: modifier,
+            rawRepresentation: group,
+          );
+
+        case _MemberType.constructor:
+          return _ConstructorMemberGroup._(
+            isNamed: isFactory || isNamed,
+            isFactory: isFactory,
+            annotation: annotation,
+            memberType: type,
+            modifier: modifier,
+            rawRepresentation: group,
+          );
+      }
+    }
+
+    return null;
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<List<_MemberInfo>> {
```

Comment:
@incendial extract visitor in separate file, and write separate tests

Reply:
@dkrutskikh extracting visitor will require to make shared models public - don't think that is a good idea. What was the reason for that suggestion?